### ISSUE TITLE
Not working on linux

### ISSUE THREAD

**Issue by dbackowski** [bjrmatos/electron-html-to#87]
Example from readme:

``` javascript
var fs = require('fs'),
    convertFactory = require('electron-html-to');

var conversion = convertFactory({
  converterPath: convertFactory.converters.PDF,
  strategy: 'electron-server'
});

conversion({ html: '<h1>Hello World</h1>' }, function(err, result) {
  if (err) {
    return console.error(err);
  }

  console.log(result.numberOfPages);
  result.stream.pipe(fs.createWriteStream('test.pdf'));
  conversion.kill();
});
```

 doesn't work on linux:

```
{ [Error: electron manager could not start all workers..]
  workerErrors: 
   [ [Error: Unable to reach electron worker - mode: server],
     [Error: Unable to reach electron worker - mode: server] ] }
```

**Comment by bjrmatos** [comment]
@dbackowski yes, this is a limitation of electron, since it needs a GUI to be able to do its magic, but there are ways to be able to run electron inside linux (basically using xvfb) but i need to test for myself and see what problems occurs, [see this SO question for more details](http://stackoverflow.com/a/34796943/4111743)

**Comment by dbackowski** [comment]
@bjrmatos yes i known this needs a GUI, I wrote a simple markdown editor in electron that use electron-html-to to produce pdf export from html.
On Mac OS X this works fine, on Linux this isn't working and got message "electron manager could not start all workers..".
This is my app https://github.com/dbackowski/md-editor

**Comment by bjrmatos** [comment]
Great! Let me find some time to find the problem in linux and document the necessary steps to make it work!

**Comment by dbackowski** [comment]
Thx :smiley:

**Comment by albinekb** [comment]
Any update on this? @bjrmatos

**Comment by pofider** [comment]
FYI I've tried running it on linux and it truly doesn't. 

The trouble is not the pdf rendering which can be easily fixed with xvfb, but the IPC happening in the particular implementation of this package. The IPC communication between node and electron process doesn't seem to come through on linux. I've submitted question to the electron community, I'll ping back when I've some more information.

**Comment by pofider** [comment]
The link to the question describing the problem https://discuss.atom.io/t/node-js-to-electron-ipc-on-linux/31820

**Comment by pofider** [comment]
I've tried another channel to get an attention to this topic - https://github.com/electron/electron/issues/6868

**Comment by bjrmatos** [comment]
hi @pofider , thnks for your time in this issue.

i was able to get electron ipc working in ubuntu 16.04 (using your example) 

electron-prebuilt 1.3.4
node 4.4.7

steps to make it work:
- make sure to have installed the following libraries:

`sudo apt-get install build-essential clang libdbus-1-dev libgtk2.0-dev libnotify-dev libgnome-keyring-dev libgconf2-dev libasound2-dev libcap-dev libcups2-dev libxtst-dev libxss1 libnss3-dev gcc-multilib g++-multilib curl gperf bison` ([listed in electron page](https://github.com/electron/electron/blob/master/docs/development/build-instructions-linux.md#build-instructions-linux)) (i'm sure some of them are not necessary for final consumers but you know.. just in case 😄 )

`sudo apt-get install libgtkextra-dev libnss3 libasound2` ([listed here](https://github.com/electron-userland/electron-prebuilt/issues/92#issuecomment-181895095))

`sudo apt-get install -y libgtk2.0-0 libnotify-bin libgconf-2-4 libnss3 xvfb` to run xvfb (as detailed in this [blog post](https://fraserxu.me/2016/01/29/setup-electron-on-ubuntu/))
- run 

`export DISPLAY=':99.0'`
`Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &`

then just test it with `node index.js`

``` js
// index.js
var electron = require('electron-prebuilt')
var spawn = require('child_process').spawn

var child = spawn(electron, ['electron-script.js'], {
  stdio: ['ipc']
})

// this line is very helpful, it will tell you if there are missing dependencies in electron binary
child.stderr.pipe(process.stderr);

child.on('message', function (m) {
  console.log('Yes it works!')
})
```

``` js
// electron-script.js

process.send('I am ok')
process.exit(0);
```

can you verify if it works for you on ubuntu 14? maybe there is a missing dependency in your PC

make sure that `/path/to/electron-prebuilt/dist/electron --help` gives you some output first, if not there are problems in electron startup

**Comment by bjrmatos** [comment]
now i'm testing `electron-html-to` and `electron-workers` in ubuntu, lets see how it goes..

**Comment by pofider** [comment]
@bjrmatos You rock!  The IPC works after installing the libraries you mentioned. 

The `electron-html-to` doesn't work for me still, but we are definitely one step closer.

**Comment by bjrmatos** [comment]
@pofider aha! i've found the problem.. in linux if i pass the `env` option for `child_process.spawn`, the electron process crash immediately. can you verify that? just comment the env option in [this line](https://github.com/bjrmatos/electron-workers/blob/master/src/ElectronWorker.js#L199), (it will crash even if you pass an empty `env` object, `env: {}`) seems like the entire `env` for the electron process is overridden and that's why it ended up crashed, definitely an issue in electron linux. 

i will post the details in the electron issue you have opened

**Comment by destegabry** [comment]
+1 on this, I would like to test electron-html-to in a Debian pre-production server... now I'm getting some sort of muted error in the convert callback:

``` javascript
    error: {
      "workerErrors": [
        {},
        {}
      ]
    }
```

**Comment by craigbroadbear** [comment]
I am getting the same issue on Windows 10 when I try to run the example code: "electron manager could not start all workers"...

**Comment by craigbroadbear** [comment]
Ok, problem fixed.  I ran: 
 npm install electron-prebuilt

And all was well.

**Comment by pofider** [comment]
The workaround mentioned [here](https://github.com/electron/electron/issues/6868#issuecomment-252423414) seems to help. I'll have pull request tomorrow.

**Comment by bjrmatos** [comment]
fixed in `v2.2.0`
